src/averageAges.js

```diff
@@ -37,7 +41,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const hoIsWoman = people.filter(person => person.sex === 'f');
+
+  const womenFiltr = !withChildren
+    ? hoIsWoman
+    : hoIsWoman.filter(
+      mathers => people.find(child => child.mother === mathers.name)
+    );
+
+  const womenAverageAge = womenFiltr.reduce((sum, women) => (
+    sum + women.died - women.born
+  ), 0) / womenFiltr.length;
+
```

Comment:
You do the same on `calculateMenAverageAge`. You can create a separate function and use it here and in `calculateMenAverageAge`.

As an option(up to you) you could create 2 separate variables: `women` and `womenWithChildren` and pass them as arguments to call your new `getAverage` function. This way ternary operator would be in return statement and code would be more clear.